core/silkworm/execution/processor.cpp

```diff
@@ -149,21 +149,22 @@ uint64_t ExecutionProcessor::refund_gas(const Transaction& txn, uint64_t gas_lef
 }
 
 ValidationResult ExecutionProcessor::execute_block_no_post_validation(std::vector<Receipt>& receipts) noexcept {
-    receipts.clear();
-    receipts.reserve(evm_.block().transactions.size());
+    const size_t n{evm_.block().transactions.size()};
+    receipts.resize(n);
 
     uint64_t block_num{evm_.block().header.number};
     if (block_num == evm_.config().dao_block) {
         dao::transfer_balances(state_);
     }
 
     cumulative_gas_used_ = 0;
-    for (const Transaction& txn : evm_.block().transactions) {
+    for (size_t i{0}; i < n; ++i) {
+        const Transaction& txn{evm_.block().transactions[i]};
         const ValidationResult err{validate_transaction(txn)};
         if (err != ValidationResult::kOk) {
             return err;
         }
-        receipts.push_back(execute_transaction(txn));
+        receipts[i] = execute_transaction(txn);
     }
```

Comment:
Why not
```suggestion
    receipts.resize(evm_.block().transactions.size());
    auto receipts_it{receipts.begin()};
    
    uint64_t block_num{evm_.block().header.number};
    if (block_num == evm_.config().dao_block) {
        dao::transfer_balances(state_);
    }
    
    cumulative_gas_used_ = 0;
    for (const auto& txn : evm_.block().transactions) {
        if (auto err{validate_transaction(txn)} ; err != ValidationResult::kOk) {
            return err;
        }
        *receipts_it = execute_transaction(txn);
        ++receipts_it;
    }
```

Reply:
Well, I find the index version simpler and more readable than the iterator version, but that's a matter of taste.

Reply:
Imho is the opposite ... however is only a suggestion